pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -2119,3 +2137,72 @@ func getRouteHostname(obj interface{}) (host string) {
 	}
 	return host
 }
+
+func checkAndCreateAdminSecret(service kubernetes.PlatformService, namespace string, secretName string, environment api.EnvironmentType) (error, string, string) {
+	/* The default secret will be like this
+	apiVersion: v1
+	kind: Secret
+	metadata:
+	  name: kie-admin-credentials
+	type: Opaque
+	data:
+	  //adminUser
+	  username: YWRtaW4=
+	  //RedHat
+	  password: UmVkSGF0
+	*/
+
+	providedSecret, err := getSecret(service, namespace, secretName)
+
+	// The secret passed by the user doesn't exist we create one with default values, except for the trial env
+	if err != nil {
+		username := constants.DefaultAdminUser
+		password := constants.DefaultPassword
+		if !strings.HasSuffix(string(environment), constants.TrialEnvSuffix) {
+			password = string(shared.GeneratePassword(8))
+		}
+
+		err := createSecret(service, namespace, constants.KIE_ADMIN_CREDENTIALS_SECRET, username, password)
+		if err != nil {
+			log.Error("Can't create Admin Secret. ", err)
+			return errors.New("Isn't possible to create a secret with default values"), "", ""
+		} else {
+			return nil, username, password
+		}
+	} else {
+		// the customer secret exists and we read the keys to see if is correct
+		username := providedSecret.StringData[constants.USERNAME_ADMIN_SECRET_KEY]
+		password := providedSecret.StringData[constants.PASSWORD_ADMIN_SECRET_KEY]
+		if len(username) > 0 && len(password) > 0 {
+			log.Info("Found " + secretName + " secret")
+		} else {
+			return errors.New("Found" + secretName + " but lack username or password "), "", ""
+		}
+		return nil, username, password
+	}
+}
+
+func createSecret(service kubernetes.PlatformService, namespace string, secretName string, username string, password string) error {
+	secret := corev1.Secret{
```

Comment:
for secrets created by the operator, add the ownerReference so it gets deleted when the kieapp is deleted as well.

Reply:
Added